examples/manual_encrypted_verify.py

```diff
@@ -0,0 +1,199 @@
+    "URDEVICEID",
+    ]
+
+# the ID of the room you want your bot to join and send commands in.
+# This can be a direct message or room; Matrix treats them the same
+ROOM_ID = "!myfavouriteroom:example.org"
+
+ALICE_USER_ID = "@alice:example.org"
+ALICE_HOMESERVER = "https://matrix.example.org"
+ALICE_PASSWORD = "hunter2"
+
+def write_details_to_disk(resp: LoginResponse) -> None:
+    """Writes login details to disk so that we can restore our session later
+    without logging in again and creating a new device ID.
+
+    Arguments:
+        resp {LoginResponse} -- the successful client login response.
+    """
+    with open(SESSION_DETAILS_FILE, "w") as f:
+        json.dump({
+            "access_token": resp.access_token,
+            "device_id": resp.device_id,
+            "user_id": resp.user_id
+        }, f)
+
+
+async def main() -> None:
+    """A very simple encrypted nio application.
+    """
+    # If the store location doesn't exist, we'll make it
+    if not os.path.isdir(STORE_FOLDER):
+        os.mkdir(STORE_FOLDER)
+
+    # By setting `store_sync_tokens` to true, we'll save sync tokens to our
+    # store every time we sync, thereby preventing reading old, previously read
+    # events on each new sync.
+    # For more info, check out https://matrix-nio.readthedocs.io/en/latest/nio.html#asyncclient
+    config = ClientConfig(store_sync_tokens=True)
+    client = AsyncClient(
+        ALICE_HOMESERVER,
+        ALICE_USER_ID,
+        store_path=STORE_FOLDER,
+        config=config
+    )
+
+    # restore the previous session if we can, otherwise login
+    if os.path.exists(SESSION_DETAILS_FILE) and os.path.isfile(SESSION_DETAILS_FILE):
+        try:
+            with open(SESSION_DETAILS_FILE, "r") as f:
+                config = json.load(f)
+                client.access_token = config['access_token']
+                client.user_id = config['user_id']
+                client.device_id = config['device_id']
+
+                # This loads our verified/blacklisted devices and our keys
+                client.load_store()
+                print(f"Logged in using stored credentials: {client.user_id} on {client.device_id}")
+
+        except IOError as err:
+            print(f"Couldn't load session from file. Logging in. Error: {err}")
+        except json.JSONDecodeError as err:
+            print(f"Couldn't read JSON file; overwriting")
+
+    if not client.user_id or not client.access_token or not client.device_id:
+        resp = await client.login(ALICE_PASSWORD)
+        if isinstance(resp, LoginResponse):
+            print("Logged in using a password; saving details to disk")
+            write_details_to_disk(resp)
+        else:
+            print(f"Failed to log in: {resp}")
+            sys.exit(1)
+
+    # If your room isn't public, you'll need to invite this user before
+    # it joins the room
+    await client.join(ROOM_ID)
+    await client.sync(full_state=True)
```

Comment:
I think we should change this so `sync_forever()` runs in a [task](https://docs.python.org/3/library/asyncio-task.html#creating-tasks). `sync_forever()` does more than just syncing and it's needed for encryption to work correctly.

We should explain why `full_state=True` is important.

We can await the first sync using [this](https://matrix-nio.readthedocs.io/en/latest/nio.html#nio.AsyncClient.synced) asyncio event with `await client.synced.wait()`.